packages/api/src/add-build.js

```diff
@@ -90,11 +93,13 @@ async function addBuild (testCases, buildInfo, client, collectionName = 'reposit
     // 2: Query an appropriate amount of test runs, for now 100 most recent
     var snapshot = await dbRepo.collection('tests').doc(testCase.encodedName).collection('runs').orderBy('timestamp').limit(100).get();
 
-    // 3: Compute the passing percent and write that back
-    await dbRepo.collection('tests').doc(testCase.encodedName).update({
-      environments: Firestore.FieldValue.arrayUnion(buildInfo.environment),
-      percentpassing: testCasePassingPercent(snapshot)
-    }, { merge: true });
+    // 3: Compute the passing percent and write that back as well as updated environment variables
```

Comment:
I appreciate the comments outlining your algorithms, this is exactly the sort of thing comments are great for.